contrib/timelapse.lua

```diff
@@ -0,0 +1,463 @@
+  table.unpack(format_list)
+}
+
+local destination_label = dt.new_widget('section_label'){
+  label = _('output file destination'),
+  tooltip = _('settings of output file destination and name')
+}
+
+local output_directory_chooser = dt.new_widget('file_chooser_button'){
+  title = _('Select export path'),
+  is_directory = true,
+  tooltip =_('select the target directory for the timelapse. \nthe filename is created automatically.'),
+  value = string_pref_read("export_path", os.getenv('HOME')),
+  changed_callback = string_pref_write("export_path")
+}
+
+local auto_output_directory_btn = dt.new_widget('check_button') {
+  label = '',
+  tooltip = _('if selected, output video will be placed in the same directory as first of selected images'),
+  value = not dt.preferences.read(MODULE_NAME, "not_auto_output_directory", "bool"), -- reverse, for true as default
+  clicked_callback = function (widget)
+    dt.preferences.write(MODULE_NAME, "not_auto_output_directory", "bool",  not widget.value)
+    output_directory_chooser.sensitive = not output_directory_chooser.sensitive 
+  end
+}
+
+local destination_box = dt.new_widget('box') {
+  orientation = 'horizontal',
+  auto_output_directory_btn,
+  output_directory_chooser
+}
+
+local override_output_cb = dt.new_widget('check_button'){
+  label = _('override output file on conflict'),
+  tooltip = _('if checked, in case of file name conflict, the file will be overwritten'),
+  value = dt.preferences.read(MODULE_NAME, "override_output", "bool"),
+  clicked_callback = function (widget)
+    dt.preferences.write(MODULE_NAME, "override_output", "bool",  widget.value)
+  end
+}
+
+local filename_entry = dt.new_widget('entry'){
+  tooltip = _("enter output file name without extension.\n\n".. 
+    "You can use some placeholders:\n"..
+    "- {time} - time in format HH-mm-ss\n"..
+    "- {date} - date in foramt YYYY-mm-dd\n"..
+    "- {first_file} - name of first input file\n"..
+    "- {last_file} - name of last last_file"
+    ),
+  text = string_pref_read("filename_entry","timelapse_{date}_{time}"),
+ -- changed_callback = string_pref_write("filename_entry", "text") -- Not imlemented yet
+}
+
+
+local output_box = dt.new_widget('box'){
+  orientation='vertical',
+  destination_label,
+  override_output_cb,
+  destination_box,
+  filename_entry,
+}
+
+local open_after_export_cb = dt.new_widget('check_button'){
+  label = _(' open after export'),
+  tooltip = _('open video file after successful export'),
+  value = dt.preferences.read(MODULE_NAME, "open_after_export", "bool"),
+  clicked_callback = function (widget)
+    dt.preferences.write(MODULE_NAME, "open_after_export", "bool",  widget.value)
+  end
+}
+
+local module_widget = dt.new_widget('box') {
+  orientation = 'vertical',
+  res_selector,
+  framerates_selector,
+  format_selector,
+  codec_selector,
+  output_box,
+  open_after_export_cb
+}
+
+---- EXPORT & REGISTRATION
+
+local function support_format(storage, format)
+  return true
+end
+
```

Comment:
You really don't need this function unless you are going to restrict the formats that are available to the user.  You can remove this and just put nil for the argument to dt.register_storage